### ISSUE TITLE
No external function calls

### ISSUE THREAD

**Issue by j-hui** [shux-lang/shux#16]
Right now, our language _cannot_ interface with any outside functions. We need to design the syntax to do this *ASAP*, in time for our Hello World demo.

I propose that we do something like the following (for a `print()` function wrapper):
```
extern print(string);
extern print_int(int);
extern print_float(float);

etc.
```
and then we write C wrappers for them:
```
void shux_print(char *str)
{
    printf("%s", str);
}

void shux_print_int(int d)
{
    printf("%d", i);
}

void shux_print_float(float f)
{
    printf("%f", f);
}
```
which we will then link with our library. This might be tricky for vectors though.

**Comment by j-hui** [comment]
In order to wrap other library calls, for example `sleep()`, we would do the following in shux:
```
extern int sleep(int seconds);
```
and then link it to the following C wrapper:
```
int shux_sleep(int seconds)
{
    return sleep(seconds);
}
```

**Comment by j-hui** [comment]
Fixed in #17

**Comment by j-hui** [comment]
proposal printing for printing our finite-sized vectors

```
extern print_vec3(vector<3> v);
```

```
int shux_print_vec3(float[3] v)
{
    printf("<%f, %f, %f>", v[0], v[1], v[2]);
}
```
